### ISSUE TITLE
Error adding 1000 genomes variant set

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#946]
I used the repo manager to add 1000 genomes VCF to a dataset: `python repo_dev.py add-variantset ga4gh-example-data brca1 ../1000genomes/vcf --moveMode link`.

After rebooting the server I got this error:

```
➜  server-1 git:(va_fixes) ✗ python server_dev.py
Traceback (most recent call last):
  File "server_dev.py", line 7, in <module>
    ga4gh.cli.server_main()
  File "/Users/david/server-1/ga4gh/cli.py", line 89, in server_main
    parsedArgs.config_file, parsedArgs.config, parsedArgs.port)
  File "/Users/david/server-1/ga4gh/frontend.py", line 212, in configure
    dataSource.netloc, dataSource.path))
  File "/Users/david/server-1/ga4gh/datarepo.py", line 201, in __init__
    objectAdder(constructor(setName, relativePath, self))
  File "/Users/david/server-1/ga4gh/datamodel/datasets.py", line 216, in __init__
    self, localId, relativePath, dataRepository)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 538, in __init__
    self._scanDataFiles(dataDir, ['*.bcf', '*.vcf.gz'])
  File "/Users/david/server-1/ga4gh/datamodel/__init__.py", line 471, in _scanDataFiles
    self._addDataFile(filename)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 594, in _addDataFile
    self._updateMetadata(varFile)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 552, in _updateMetadata
    variantFile.filename)
ga4gh.exceptions.InconsistentMetaDataException: Metadata of ga4gh-example-data/datasets/brca1/variants/vcf/ALL.chrMT.phase3_callmom.20130502.genotypes.vcf.gz is not consistent with other files in this VariantSet, and inconsistent metadata is not permitted
```

**Comment by david4096** [comment]
Which also causes list to fail making it necessary to remove it by hand:

```
python repo_dev.py list ga4gh-example-data                                                   
Repo at 'ga4gh-example-data'
Listing
Traceback (most recent call last):
  File "repo_dev.py", line 7, in <module>
    ga4gh.cli.repo_main()
  File "/Users/david/server-1/ga4gh/cli.py", line 1606, in repo_main
    runner.run()
  File "/Users/david/server-1/ga4gh/cli.py", line 1336, in run
    self.repoManager.list()
  File "/Users/david/server-1/ga4gh/repo_manager.py", line 501, in list
    self._repoPath, doConsistencyCheck=False)
  File "/Users/david/server-1/ga4gh/datarepo.py", line 201, in __init__
    objectAdder(constructor(setName, relativePath, self))
  File "/Users/david/server-1/ga4gh/datamodel/datasets.py", line 216, in __init__
    self, localId, relativePath, dataRepository)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 538, in __init__
    self._scanDataFiles(dataDir, ['*.bcf', '*.vcf.gz'])
  File "/Users/david/server-1/ga4gh/datamodel/__init__.py", line 471, in _scanDataFiles
    self._addDataFile(filename)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 594, in _addDataFile
    self._updateMetadata(varFile)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 552, in _updateMetadata
    variantFile.filename)
ga4gh.exceptions.InconsistentMetaDataException: Metadata of ga4gh-example-data/datasets/brca1/variants/vcf/ALL.chrMT.phase3_callmom.20130502.genotypes.vcf.gz is not consistent with other files in this VariantSet, and inconsistent metadata is not permitted
```

**Comment by jeromekelleher** [comment]
Good catch, thanks @david4096.

**Comment by dcolligan** [comment]
@david4096 where is the data you are using, just so I can reproduce?

**Comment by david4096** [comment]
ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/

**Comment by david4096** [comment]
If we're checking for the presence of a reference for read groups, shouldn't we do the same for variants?

**Comment by diekhans** [comment]
@david4096 good point.  While technical the API should be able to server up both reads and variants without the underlying references, the behavior is left unspecified.  However, it's silly not to have the reference sequences and it's a reasonable restriction for the reference server to place on the data it hosts. It reduces data load errors.  If so, it should be done consistently for reads, variants and sequence annotations.

**Comment by dcolligan** [comment]
Fixed in #949
